### PR TITLE
Single Service call for all the pagination params

### PR DESCRIPTION
**Pull request by srikanthv02**
@revinjacob : 

a)The pending service call has been implemented for calling the pagination params. I haven't removed Hrushikesh's file(uiGridOptionsServices.js) . I have changed my service name and using it accordingly from PaginationUtility file. 

b) There was one more bug which has been resolved related to the new task being created when the page size has been changed.

### CODE REVIEW

File: client/cat3/src/partials/sections/dashboard/workzone/azureARM/azureARMCtrl.js [hunk 1]
```diff
@@ -8,16 +8,12 @@
...
-			$scope.paginationParams = {
-				pages: {
-					page: 1,
-					pageSize: 1
-				}
-			};
-			$scope.currentCardPage = $scope.paginationParams.pages.page;
-			$scope.cardsPerPage = $scope.paginationParams.pages.pageSize;
+			var armData = uiGridOptionsService.options();
+			$scope.paginationParams = armData.pagination;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/azureARM/azureARMCtrl.js]: Change variable armData appropriately to denote what it contains. It does not only uigrid defaults

File: client/cat3/src/partials/sections/dashboard/workzone/cloudFormation/cloudFormationCtrl.js [hunk 2]
```diff
@@ -13,16 +13,12 @@
...
-					pageSize: 1
-				}
-			};
-			$scope.currentCardPage = $scope.paginationParams.pages.page;
-			$scope.cardsPerPage = $scope.paginationParams.pages.pageSize;
+			var cftData = uiGridOptionsService.options();
+			$scope.paginationParams = cftData.pagination;
+			$scope.currentCardPage = cftData.pagination.page;
+			$scope.cardsPerPage = cftData.pagination.pageSize;
 			$scope.numofCardPages = 0; //Have to calculate from totalItems/cardsPerPage
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/cloudFormation/cloudFormationCtrl.js]: Change variable cftData appropriately to denote what it contains. It contains uigrid defaults

File: client/cat3/src/partials/sections/dashboard/workzone/container/containerCtrl.js [hunk 3]
```diff
@@ -8,71 +8,52 @@
...
-				},
-				sort:{
-					field:'',
-					direction:''
-				}
-			};
-
+			var containerData = uiGridOptionsService.options();
+			$scope.paginationParams = containerData.pagination;
 			$scope.tabData = [];
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/container/containerCtrl.js]: Change variable containerData appropriately to denote what it contains. It contains uigrid defaults

File: client/cat3/src/partials/sections/dashboard/workzone/container/containerCtrl.js [hunk 4]
```diff
@@ -8,71 +8,52 @@
...
-				gridApi.pagination.on.paginationChanged($scope, function(newPage, pageSize) {
-					$scope.paginationParams.pages = {
-						page: newPage,
-						pageSize: pageSize
-					};
-					$scope.getContainerList();
-				});
-			};
+			$scope.containerGridOptions = angular.extend(containerData.gridOption, {
+				onRegisterApi :function(gridApi) {
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/container/containerCtrl.js]: Extending two times, but each is a fresh assignment to $scope.containerGridOptions. Please recheck.

File: client/cat3/src/partials/sections/dashboard/workzone/orchestration/orchestrationCtrl.js [hunk 5]
```diff
@@ -17,81 +17,63 @@
...
+					$scope.gridApi.core.on.sortChanged($scope, function(grid, sortColumns) {
+						if (sortColumns[0] && sortColumns[0].field && sortColumns[0].sort && sortColumns[0].sort.direction) {
+							$scope.paginationParams.sortBy = sortColumns[0].field;
+							$scope.paginationParams.sortOrder = sortColumns[0].sort.direction;
+							$scope.taskListGridView();
+						}
+					});
+					//Pagination for page and pageSize
+					gridApi.pagination.on.paginationChanged($scope, function(newPage, pageSize) {
+						$scope.paginationParams.page = newPage;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/orchestration/orchestrationCtrl.js]: Use gridApi consistently. Line 62 uses $scope.gridApi, line 71 uses gridApi. Avoid usage of $scope.gridApi if possible.

File: client/cat3/src/utility/paginationUtility.js [hunk 6]
```diff
@@ -6,19 +6,42 @@
...
+                        useExternalPagination: true,
+                        useExternalSorting: true,
+                        multiSelect: false,
+                        modifierKeysToMultiSelect: false,
+                        enableRowSelection: true,
+                        enableRowHeaderSelection: false
+                    }
+                };
+            }
+        }
```
Comment (revinjacob) [inline client/cat3/src/utility/paginationUtility.js]: This implementation of defaults is specific to server side paginated ui-grid tables. The utility.pagination module should provide defaults for client side paginated ui-grid tables as well.

File: client/cat3/src/utility/paginationUtility.js [hunk 7]
```diff
@@ -6,19 +6,42 @@
...
+    angular.module('utility.pagination', []).service('uiGridOptionsService', [function(){
+        return {
+            options: function() {
+                return {
+                    pagination: {
+                        pageSize: 10,
+                        page: 1,
+                        sortBy: "",
+                        sortOrder: ""
+                    },
```
Comment (revinjacob) [inline client/cat3/src/utility/paginationUtility.js]: The defaults are specifically for ui-grid usage. Object format of defaults provided to controller is different from actual values it expects back from controller. Use consistent object format for ui-grid pagination object.

Any reason why the format is changed FROM 
{
	pages:{
		page:'',
		pageSize:''
	},
	sort:{
		field:'',
		direction:''
	}
}
TO
{
	pageSize:''
	page:'',
	sortBy:'',
	sortOrder:''
}

File: client/cat3/src/partials/sections/dashboard/workzone/orchestration/orchestrationCtrl.js [hunk 8]
```diff
@@ -299,15 +275,18 @@
...
 			$scope.setLastPageView = function(){
 				$scope.orcheGridOptions.totalItems = $scope.orcheGridOptions.totalItems + 1;
 				var lastPage = (Math.ceil(($scope.orcheGridOptions.totalItems)/$scope.orcheGridOptions.paginationPageSize));
 				//Set sortBy and sortField to controller defaults;
 				helper.setPaginationDefaults();
-				$scope.orcheGridOptions.paginationCurrentPage = $scope.paginationParams.pages.page = lastPage;				
+				/*setting the pageSize value(by default it takes 10 which 
+				will break when we change the pageSize value in the grid)*/
+				$scope.paginationParams.pageSize = $scope.orcheGridOptions.paginationPageSize;
+				$scope.orcheGridOptions.paginationCurrentPage = $scope.paginationParams.page = lastPage;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/orchestration/orchestrationCtrl.js]: Dont read from $scope.orcheGridOptions.paginationPageSize, instead use $scope.paginationParams.pageSize as its unchanged

**Comment by srikanthv02** [comment]
@revinjacob : Review comments have been taken care and the changes have been done. Have left 2 points for @arabind (defaults for ui-grid usage)

**Pull request merged**